### PR TITLE
Added Higher-Order Component post

### PR DESCRIPTION
**Pull request by bloodyowl**

### CODE REVIEW

File: content/fr/articles/js/react/higher-order-component/index.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+title: Les patterns Provider & Higher-Order Component avec React
+tags:
+  - javascript
+  - react
+  - higher-order-component
+authors:
+  - bloodyowl
+---
+
+# The Provider and Higher-Order Component patterns with React
```
Comment (MoOx) [inline content/fr/articles/js/react/higher-order-component/index.md]: remove this line

File: content/fr/articles/js/react/higher-order-component/index.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
+---
+date: "2016-06-08"
```
Comment (MoOx) [inline content/fr/articles/js/react/higher-order-component/index.md]: maybe next tuesday? To avoid LOT OF POSTS, then NOTHING, then LOT OF POSTS :)

Comment (MoOx) [inline content/fr/articles/js/react/higher-order-component/index.md]: Cool !
But I think it's better to plan 1 per week, than 4 in 2 weeks, then nothing.
We might reconsider this when your PR queue will have more than 4 posts

**Comment by MoOx** [comment]
Except for the date, lgtm, nice work @bloodyowl!

File: content/fr/articles/js/react/higher-order-component/index.md (hunk 2 above)
Comment (MoOx) [inline content/fr/articles/js/react/higher-order-component/index.md]: If you are really entusiatic, we can do each tuesday and thrusday.

Comment (MoOx) [inline content/fr/articles/js/react/higher-order-component/index.md]: If have to admit that I have like 4 o 5 posts almost ready to ship... ^^

**Comment by magsout** [comment]
lgtm... Love this post...

**Comment by lionelB** [comment]
Could be nice to add this video at the end of the post https://www.youtube.com/watch?v=zD_judE-bXk

All about Hoc, composition and some usefull tip (like using a `compose()`  to merge several Hoc into one )

**Comment by magsout** [comment]
@lionelB I didn't know `compose()` thanks ;)

**Comment by lionelB** [comment]
l'avantage aussi avec compose c'est que ton source est plus clean 
```jsx
<A>
  <B>
    <C>
      <Component />
    </C>
  </B>
</A>
```
mais 
```
<Composed>
  <Component />
</Composed>
```

**Pull request merged**